java/code/src/com/redhat/rhn/domain/action/ActionFactory.java

```diff
@@ -950,6 +950,23 @@ public static void removeInvalidResults(Action action) {
         }
     }
 
+    /**
+     * Update the status of several rhnServerAction rows identified by server and action IDs.
+     * @param actionIn associated action of rhnServerAction records
+     * @param serverIds server Ids for which action is scheduled
+     * @param status {@link ActionStatus} object that needs to be set
+     */
+    public static void updateServerActionsPickedUp(Action actionIn, List<Long> serverIds, ActionStatus status) {
```

Comment:
I do not see why this function must have `ActionStatus` as input parameter?
The only sense of this is to set it to PickedUp, so we can hardcode it.

I made a PR to change this.

Reply:
PR merged into this one, thank you

Reply:
I see you already implemented the change and it's pending in https://github.com/SUSE/spacewalk/pull/16285/commits . So I will backport it only to 4.2 then. Is that okay?

Reply:
https://github.com/SUSE/spacewalk/pull/16298

Reply:
I have it already in https://github.com/SUSE/spacewalk/pull/16286. Let's do a race who merge first :-)